src/Iterable/windowed.php

```diff
@@ -41,9 +41,10 @@ function windowed(
         if ($size > $windowSize) {
             break;
         }
-        $result[] = $transform(array_slice(iterator_to_array($iterable), $index, $windowSize));
+        $array = is_array($iterable) ? $iterable : iterator_to_array($iterable);
```

Comment:
I'd rather do it outside the `while()` loop otherwise this operation will take place multiple times, and it is quite slow.

Reply:
Good point, my friend 👏🏻 

Consider it done.